src/UI/Implementation/Component/Modal/Renderer.php

```diff
@@ -47,27 +47,27 @@ public function registerResources(ResourceRegistry $registry) {
 	 * @param Component\Modal\Modal $modal
 	 * @param string $id
 	 */
-	protected function registerSignals(Component\Modal\Modal $modal, $id) {
+	protected function registerSignals(Component\Modal\Modal $modal) {
 		$show = $modal->getShowSignal();
 		$close = $modal->getCloseSignal();
-		$js = $this->getJavascriptBinding();
```

Comment:
Maybe explain in the comment of this function why this pattern here:

```
$js = $this->getJavascriptBinding();
$js->addOnLoadCode("
 			$(document).on('{$show}', function() { il.UI.modal.showModal('{$id}', {$options}); return false; });
 			$(document).on('{$close}', function() { il.UI.modal.closeModal('{$id}'); return false; });"
);
```
Is bad and should not be used and refer to this pattern:		
```
return $modal->withAdditionalOnLoadCode(function($id) use (...) {
			return ...
});
```

To be used instead.